src/features/common/Layout/Header/accountHeader.tsx

```diff
@@ -0,0 +1,138 @@
+import React,{ useEffect } from 'react';
+import styles from './Header.module.scss';
+import { useRouter } from 'next/router';
+
+import Settings from '../../../../../public/assets/images/icons/userProfileIcons/Settings';
+import SettingsModal from '../../../user/UserProfile/components/SettingsModal';
+import { useAuth0 } from '@auth0/auth0-react';
+import { getRequest, getAccountInfo } from '../../../../utils/apiRequests/api';
+import {
+  setUserExistsInDB,
+  removeUserExistsInDB,
+  getLocalUserInfo,
+  removeLocalUserInfo,
+} from '../../../../utils/auth0/localStorageUtils';
+import BackButton from '../../../../../public/assets/images/icons/BackButton';
+import i18next from '../../../../../i18n'
+const { useTranslation } = i18next;
+
+export default function AcountHeader(props: any) {
+  const [userprofile, setUserprofile] = React.useState();
+  const [forceReload, changeForceReload] = React.useState(false);
+  const { t } = useTranslation(['me']);
+
+  const {
+    isLoading,
+    isAuthenticated,
+    loginWithRedirect,
+    getAccessTokenSilently,
+    logout
+  } = useAuth0();
+  const router = useRouter();
+  const logoutUser = () => {
+    removeLocalUserInfo();
+    logout({ returnTo: `${process.env.NEXTAUTH_URL}/` });
+  }
+
+  React.useEffect(() => {
+    async function loadUserData() {
+      if (typeof Storage !== 'undefined') {
+        let token = null;
+        if (isAuthenticated) {
+          token = await getAccessTokenSilently();
+        }
+        if (!isLoading && token) {
+          try {
+            const res = await getAccountInfo(token)
+            if (res.status === 200) {
+              const resJson = await res.json();
+              setUserprofile(resJson);
+            } else if (res.status === 303) {
+              // if 303 -> user doesn not exist in db
+              setUserExistsInDB(false)
+              if (typeof window !== 'undefined') {
+                router.push('/complete-signup');
+              }
+            } else if (res.status === 401) {
+              // in case of 401 - invalid token: signIn()
+              logoutUser();
+              removeUserExistsInDB()
+              loginWithRedirect({redirectUri:`${process.env.NEXTAUTH_URL}/login`, ui_locales: localStorage.getItem('language') || 'en' });
+            } else {
+              // any other error
+            }
+          } catch (err) {
+            console.log(err);
+          }
+        } else {
+          loginWithRedirect({
+            redirectUri: `${process.env.NEXTAUTH_URL}/account/history`,
+            ui_locales: localStorage.getItem('language') || 'en',
+          });
```

Comment:
This code block looks like a duplicate the following code - can't we abstract it somehow? https://github.com/Plant-for-the-Planet-org/planet-webapp/blob/f358ad2e551ba81584c090d0ca88c105c59f0b09/src/features/user/Account/screens/index.tsx#L73-L104  A similar code pattern also is used in https://github.com/Plant-for-the-Planet-org/planet-webapp/blob/develop/src/utils/auth0/userInfo.ts.

Reply:
Yess I forgot to remove it from accounts

Reply:
@sarveshpro can we remove it from accounts?

Reply:
I'll suggest you create a userContext. that way you can update it's value from here and use it anywhere in the app.